staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go

```diff
@@ -1880,3 +1880,72 @@ func (l *fakeNodeLister) Get(name string) (*v1.Node, error) {
 	}
 	return nil, nil
 }
+
+func TestLoadBalancerClass(t *testing.T) {
+	testcases := []struct {
+		name            string
+		service         *v1.Service
+		expectOp        loadBalancerOperation
+		expectCloudCall bool
+	}{
+		{
+			name: "create a load balancer type of service with loadBalancerClass",
+			service: &v1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-load-balancer-class",
+					Namespace: "default",
+				},
+				Spec: v1.ServiceSpec{
+					Type: v1.ServiceTypeLoadBalancer,
+					Ports: []v1.ServicePort{{
+						Port: int32(80),
+					}},
+					LoadBalancerClass: utilpointer.StringPtr("test.com/test"),
+				},
+			},
+			expectOp:        deleteLoadBalancer,
```

Comment:
This is delete because the cloud provider no longer thinks it owns the Service?

Reply:
The main reason here is because there is only two ops: otherwise, it is `ensureLoadBalancer` or it is `deleteLoadBalancer`, in this case, it should not be `ensureLoadBalancer`, so fill it with `deleteLoadBalancer`.